cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerParcelDecommissionService.java

```diff
@@ -92,9 +92,11 @@ private Multimap<String, String> getParcelVersionsByStatusNameAndVersion(Parcels
                 .collect(Multimaps.toMultimap(ParcelInfo::getName, ParcelInfo::getVersion, HashMultimap::create));
     }
 
-    private Multimap<String, String> getUnusedParcels(Set<ParcelInfo> usedParcels, Set<String> usedParcelComponentNames, Set<String> parcelsFromImage) {
-        return usedParcels.stream()
-                .filter(usedParcel -> !usedParcelComponentNames.contains(usedParcel.getName()) && parcelsFromImage.contains(usedParcel.getName()))
+    private Multimap<String, String> getUnusedParcels(Set<ParcelInfo> availableParcels, Set<ClouderaManagerProduct> usedParcels, Set<String> parcelsFromImage) {
+        return availableParcels.stream()
+                .filter(parcel -> usedParcels.stream().noneMatch(usedParcel -> usedParcel.getName().equals(parcel.getName())
+                        && usedParcel.getVersion().equals(parcel.getVersion()))
+                        && parcelsFromImage.contains(parcel.getName()))
```

Comment:
this inner stream should be at least moved out to a variable, or even to a method. maybe logged also to help debugging